pkg/cnab/extensions/dependencies.go

```diff
@@ -51,18 +57,19 @@ type DependencyVersion struct {
 // ReadDependencies is a convenience method for returning a bonafide
 // Dependencies reference after reading from the applicable section from
 // the provided bundle
-func ReadDependencies(bun bundle.Bundle) (Dependencies, error) {
+func ReadDependencies(bun bundle.Bundle) ([]Dependency, error) {
```

Comment:
Why not return a `Dependencies` object as opposed to `[]Dependency`?

Reply:
It would be fine to return Dependencies from here, and then in solver.go ResolveDependencies immediately call `deps.ListBySequence()` and then iterate over the result.

Probably would make more sense from a separation of responsibilities to more faithfully return the full structure as represented in the bundle.json from this struct anyway.

Reply:
Changes applied  accordantly

Reply:
Now the solver is responsible to call ListBySequence() before building the DependencyLock array